contracts/src/v0.8/automation/dev/v2_3/AutomationRegistry2_3.sol

```diff
@@ -266,23 +273,30 @@ contract AutomationRegistry2_3 is AutomationRegistryBase2_3, OCR2Abstract, Chain
     uint8 f,
     OnchainConfig memory onchainConfig,
     uint64 offchainConfigVersion,
-    bytes memory offchainConfig
+    bytes memory offchainConfig,
+    address[] memory billingTokens,
+    BillingConfig[] memory billingConfigs
   ) public onlyOwner {
     if (signers.length > MAX_NUM_ORACLES) revert TooManyOracles();
     if (f == 0) revert IncorrectNumberOfFaultyOracles();
     if (signers.length != transmitters.length || signers.length <= 3 * f) revert IncorrectNumberOfSigners();
+    if (billingTokens.length != billingConfigs.length) revert ParameterLengthError();
+    // set billing config for tokens
+    _setBillingConfig(billingTokens, billingConfigs);
 
     // move all pooled payments out of the pool to each transmitter's balance
-    uint96 totalPremium = s_hotVars.totalPremium;
-    uint96 oldLength = uint96(s_transmittersList.length);
-    for (uint256 i = 0; i < oldLength; i++) {
-      _updateTransmitterBalanceFromPool(s_transmittersList[i], totalPremium, oldLength);
+    for (uint256 i = 0; i < s_transmittersList.length; i++) {
```

Comment:
no need to access `.length` so many times?

Reply:
Good Q, removed the oldLength variable to avoid stack too deep. confirmed with Ryan, a bit more gas is not a concern in this non-transmit path.